cogeo_mosaic/backend/dynamo.py

```diff
@@ -0,0 +1,135 @@
+import functools
+import itertools
+import json
+import logging
+import os
+from decimal import Decimal
+from typing import Dict, List, Optional, Tuple
+
+import boto3
+import mercantile
+from cogeo_mosaic.backend.base import BaseBackend
+from cogeo_mosaic.backend.utils import find_quadkeys, get_hash
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+class DynamoDBBackend(BaseBackend):
+    """DynamoDB Backend Adapter"""
+
+    def __init__(
+        self,
+        table_name: Optional[str] = None,
+        region: str = os.getenv("AWS_REGION", "us-east-1"),
+        load_mosaic: bool = True,
+    ):
+        self.client = boto3.resource("dynamodb", region_name=region)
+        self.table_name = table_name
+        self.mosaic_def = None
+
+        if table_name:
+            self.table = self.client.Table(table_name)
+
+            if load_mosaic:
+                self.mosaic_def = self.fetch_mosaic_definition()
```

Comment:
I'm not sure about the `load_mosaic` option, 

I think the idea is to load the mosaic when we are in a `read` mode ? 

maybe something like 
```
    def __init__(
         self,
         table_name: str,
         region: str = os.getenv("AWS_REGION", "us-east-1"),
         mosaic_def: Optional[Dict] = None,
     ):
    # Write
     if mosaic_def:
        # Here we know we should upload the mosaic_definition
        # Note: add a check if the table already exist ? 
     
     # Read
     else:
          self.table = self.client.Table(table_name)
          self.mosaic_def = self.fetch_mosaic_definition()
```

Reply:
Yeah, that's possible... I think the issue is what you do for the `create/` endpoint, when you haven't created a Mosaic yet. _Or_ you just require that a mosaic be created outside of the backends, which might be simpler

Reply:
I think I updated to an acceptable solution, where `mosaic_def` is required to be fetched from the url/bucket+key/dynamo table/file name, **or provided in `__init__`**. If neither is the case, an error is raised I think